### PR TITLE
Adding support for HTTP basic auth

### PR DESCRIPTION
**Pull request by trumant**

### CODE REVIEW

File: lib/sitemap-parser.rb [hunk 1]
```diff
@@ -11,7 +11,11 @@ def initialize(url, opts = {})
   def raw_sitemap
     @raw_sitemap ||= begin
       if @url =~ /\Ahttp/i
-        request = Typhoeus::Request.new(@url, followlocation: @options[:followlocation])
+        request_options = { followlocation: @options[:followlocation] }
+        if @options[:userpwd]
+          request_options[:userpwd] = @options[:userpwd]
+        end
+        request = Typhoeus::Request.new(@url, request_options)
```
Comment (benbalter) [inline lib/sitemap-parser.rb]: Would it make sense to simply pass @options to Typhoeus to support all Typhoeus options?

Comment (trumant) [inline lib/sitemap-parser.rb]: The only reason I did not do so is because `:recurse` is not an option we'd want to hand to Typhoeus

Comment (benbalter) [inline lib/sitemap-parser.rb]: Should we `delete` decurse before passing then and pass the rest? (Blacklist, rather than whitelist?)

**Comment by benbalter** [comment]
Nice! Thanks @trumant! Will work on getting a new release out with this.

**Pull request merged**